File: arrays_and_strings/check_for_target.java
public boolean checkForTarget(int[] nums, int target)
{
    int left = 0;
    int right = nums.length - 1;

    while (left < right)
    {
        int curr = nums[left] + nums[right];
        if (curr == target)
        {
            return true;
        }

        if (curr > target)
        {
            right--;
        }
        else
        {
            left++;
        }
    }

    return false;
}

File: arrays_and_strings/check_for_target.py
def check_for_target(nums: list[int], target: int) -> bool:
    left = 0
    right = len(nums) - 1

    while left < right:
        curr = nums[left] + nums[right]
        if curr == target:
            return True
        if curr > target:
            right -= 1
        else:
            left += 1
    
    return False